### PR TITLE
osd: Better error message when OSD count is less than osd_pool_default_size

### PR DESCRIPTION
**Pull request by ashitakasam**
osd: Better error message when OSD count is less than osd_pool_default_size

Fixes: http://tracker.ceph.com/issues/38617

Signed-off-by: zjh <jhzeng93@foxmail.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: doc/rados/operations/health-checks.rst [hunk 1]
```diff
@@ -673,6 +673,12 @@ the pool is too large and should be reduced or set to zero with::
 
 For more information, see :ref:`specifying_pool_target_size`.
 
+SMALLER_OSD_NUM
```
Comment (liewegas) [inline doc/rados/operations/health-checks.rst]: can we renamed this ``TOO_FEW_OSDS``?

File: src/mon/PGMap.cc [hunk 2]
```diff
@@ -2692,6 +2692,15 @@ void PGMap::get_health_checks(
     }
   }
 
+  // SMALLER_OSD_NUM
+  auto osd_pool_default_size = cct->_conf.get_val<uint64_t>("osd_pool_default_size");
+  if (osdmap.get_num_osds() < osd_pool_default_size) {
+    ostringstream ss;
+    ss << "OSD count " << osdmap.get_num_osds()
+	 << " < default " << osd_pool_default_size;
```
Comment (liewegas) [inline src/mon/PGMap.cc]: s/default/osd_pool_default_size/

**Comment by liewegas** [comment]
```
2019-04-30T09:41:26.132 INFO:tasks.workunit.client.0.smithi086.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:1607: wait_for_health:  ceph health detail
2019-04-30T09:41:26.482 INFO:tasks.workunit.client.0.smithi086.stdout:HEALTH_WARN OSD count 1 < osd_pool_default_size 3
2019-04-30T09:41:26.483 INFO:tasks.workunit.client.0.smithi086.stdout:TOO_FEW_OSDS OSD count 1 < osd_pool_default_size 3
```
from /a/sage-2019-04-30_01:30:03-rados-wip-sage-testing-2019-04-29-1730-distro-basic-smithi/3910482

this should fix it:
```
diff --git a/qa/standalone/ceph-helpers.sh b/qa/standalone/ceph-helpers.sh
index 9ab9ad1b851..ba79a652f1a 100755
--- a/qa/standalone/ceph-helpers.sh
+++ b/qa/standalone/ceph-helpers.sh
@@ -1626,9 +1626,11 @@ function test_wait_for_health_ok() {
     local dir=$1
 
     setup $dir || return 1
-    run_mon $dir a --osd_pool_default_size=1 --osd_failsafe_full_ratio=.99 --mon_pg_warn_min_per_osd=0 || return 1
+    run_mon $dir a --osd_failsafe_full_ratio=.99 --mon_pg_warn_min_per_osd=0 || return 1
     run_mgr $dir x --mon_pg_warn_min_per_osd=0 || return 1
     run_osd $dir 0 || return 1
+    run_osd $dir 1 || return 1
+    run_osd $dir 2 || return 1
     kill_daemons $dir TERM osd || return 1
     ceph osd down 0 || return 1
     ! TIMEOUT=1 wait_for_health_ok || return 1
```

**Comment by liewegas** [comment]
... or pass the option to ceph-mon that disables the health warning

**Comment by Liuchang0812** [comment]
jenkins test this please

File: qa/standalone/ceph-helpers.sh [hunk 3]
```diff
@@ -1626,9 +1626,11 @@ function test_wait_for_health_ok() {
     local dir=$1
 
     setup $dir || return 1
-    run_mon $dir a --osd_pool_default_size=1 --osd_failsafe_full_ratio=.99 --mon_pg_warn_min_per_osd=0 || return 1
+    run_mon $dir a --osd_failsafe_full_ratio=.99 --mon_pg_warn_min_per_osd=0 || return 1
     run_mgr $dir x --mon_pg_warn_min_per_osd=0 || return 1
     run_osd $dir 0 || return 1
+    run_osd $dir 1 || return 1
+    run_osd $dir 2 || return 1
```
Comment (tchaikov) [inline qa/standalone/ceph-helpers.sh]: this actually broke this test. i am preparing a fix.

Comment (tchaikov) [reply qa/standalone/ceph-helpers.sh]: #28328

**Pull request merged**